### ISSUE TITLE
Error Maximum call stack size exceeded

### ISSUE THREAD

**Issue by Tom910** [smelukov/webpack-runtime-analyzer#2]
When assembling error. In the project about 5000 modules
Webpack 2.1.0-beta.27

```
 58% building modules 2353/2907 modules 554 active ...f-portal-web/actions/icons/getFile.jsevents.js:96
    handler.call(self, arg1);
            ^

RangeError: Maximum call stack size exceeded
    at emitOne (events.js:96:13)
    at Sender.emit (events.js:189:7)
    at Sender.frameAndSend (/Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:212:19)
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:126:12
    at Sender.applyExtensions (/Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:287:5)
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:120:10
    at Sender.flush (/Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:268:3)
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:270:10
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:127:7
    at Sender.applyExtensions (/Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:287:5)
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:120:10
    at Sender.flush (/Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:268:3)
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:270:10
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:127:7
    at Sender.applyExtensions (/Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:287:5)
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:120:10
    at Sender.flush (/Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:268:3)
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:270:10
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:127:7
    at Sender.applyExtensions (/Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:287:5)
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:120:10
    at Sender.flush (/Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:268:3)
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:270:10
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:127:7
    at Sender.applyExtensions (/Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:287:5)
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:120:10
    at Sender.flush (/Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:268:3)
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:270:10
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:127:7
    at Sender.applyExtensions (/Volumes/files/projects/tinkoff-portal-web/node_modules/ws/lib/Sender.js:287:5)
```

**Comment by Tom910** [comment]
Reduce the number of modules is now another error

```
94% asset optimization/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack-runtime-analyzer/src/plugin/index.js:92
                    return filePath.replace(/([^?]+).*!/, '$1');
                                    ^

TypeError: filePath.replace is not a function
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/webpack-runtime-analyzer/src/plugin/index.js:92:37
    at Array.map (native)
    at RuntimeAnalyzerPlugin.<anonymous> (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack-runtime-analyzer/src/plugin/index.js:91:58)
    at Array.forEach (native)
    at RuntimeAnalyzerPlugin.<anonymous> (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack-runtime-analyzer/src/plugin/index.js:81:27)
    at Array.forEach (native)
    at RuntimeAnalyzerPlugin.<anonymous> (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack-runtime-analyzer/src/plugin/index.js:80:28)
    at next (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:140:14)
    at Compiler.<anonymous> (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack-runtime-analyzer/node_modules/webpack/lib/ProgressPlugin.js:74:4)
    at Compiler.applyPluginsAsyncSeries (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:142:13)
    at Compiler.emitAssets (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/Compiler.js:294:7)
    at onCompiled (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/Compiler.js:233:11)
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/Compiler.js:479:13
    at next (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:138:11)
    at Compiler.<anonymous> (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/CachePlugin.js:61:4)
    at Compiler.applyPluginsAsyncSeries (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:142:13)
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/Compiler.js:476:10
    at Compilation.applyPluginsAsyncSeries (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:131:46)
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/Compilation.js:627:18
    at next (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:138:11)
    at Compilation.<anonymous> (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/ProgressPlugin.js:122:5)
    at next (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:140:14)
    at Compilation.<anonymous> (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack-runtime-analyzer/node_modules/webpack/lib/ProgressPlugin.js:69:5)
    at Compilation.applyPluginsAsyncSeries (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:142:13)
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/Compilation.js:618:10
    at next (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:138:11)
    at Compilation.<anonymous> (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/ProgressPlugin.js:118:5)
    at next (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:140:14)
    at Compilation.<anonymous> (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/optimize/UglifyJsPlugin.js:144:4)
    at next (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:140:14)
```

**Comment by smelukov** [comment]
Thanks for your report.
We are working on it.

**Comment by Tom910** [comment]
If you need any more information or the debug. Write me please

**Comment by smelukov** [comment]
I've pushed an update.
Can you update webpack-runtime-analyzer to alpha4 and check to problem?

**Comment by Tom910** [comment]
If many modules, the error is the same as it was 'Maximum call stack size exceeded

If little modules, is a new bug
```
 94% asset optimization/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/Chunk.js:67
                throw new Error("Chunk.initial was removed. Use isInitial()");
                ^

Error: Chunk.initial was removed. Use isInitial()
    at Chunk.get initial [as initial] (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/Chunk.js:67:9)
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/webpack-runtime-analyzer/src/plugin/index.js:322:35
    at Array.map (native)
    at RuntimeAnalyzerPlugin.<anonymous> (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack-runtime-analyzer/src/plugin/index.js:311:40)
    at next (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:140:14)
    at Compiler.<anonymous> (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack-runtime-analyzer/node_modules/webpack/lib/ProgressPlugin.js:74:4)
    at Compiler.applyPluginsAsyncSeries (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:142:13)
    at Compiler.emitAssets (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/Compiler.js:292:7)
    at onCompiled (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/Compiler.js:231:11)
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/Compiler.js:480:13
    at next (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:138:11)
    at Compiler.<anonymous> (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/CachePlugin.js:62:5)
    at Compiler.applyPluginsAsyncSeries (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:142:13)
    at /Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/Compiler.js:477:10
    at Compilation.applyPluginsAsyncSeries (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:131:46)
    at self.applyPluginsAsync.err (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/Compilation.js:615:19)
    at next (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:138:11)
    at Compilation.<anonymous> (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/ProgressPlugin.js:122:5)
    at next (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:140:14)
        at Compilation.<anonymous> (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack-runtime-analyzer/node_modules/webpack/lib/ProgressPlugin.js:69:5)
    at Compilation.applyPluginsAsyncSeries (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:142:13)
    at self.applyPluginsAsync.err (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/Compilation.js:606:11)
    at next (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:138:11)
    at Compilation.<anonymous> (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/ProgressPlugin.js:118:5)
    at next (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:140:14)
    at Compilation.compilation.plugin (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/optimize/UglifyJsPlugin.js:140:5)
    at next (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:140:14)
    at Compilation.<anonymous> (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack-runtime-analyzer/node_modules/webpack/lib/ProgressPlugin.js:65:5)
    at Compilation.applyPluginsAsyncSeries (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/node_modules/tapable/lib/Tapable.js:142:13)
    at self.applyPluginsAsync.err (/Volumes/files/projects/tinkoff-portal-web/node_modules/webpack/lib/Compilation.js:601:10)
```

**Comment by Tom910** [comment]
By mistake, there is no compatibility even with Webpack2

**Comment by smelukov** [comment]
Yes, it seems like the plugin is not working correctly with some webpack2-builds.
Thanks for report and call stack.
I'll investigate it.

**Comment by smelukov** [comment]
@Tom910 Hi! I've pushed an update - 1.0.0-alpha5
Please update webpack-runtime-analyzer

**Comment by Tom910** [comment]
Thanks, now works.

Graph not usable http://imgur.com/a/ULDQT

**Comment by smelukov** [comment]
@Tom910 great! Thanks for issue.
Oh... realy... graph is not usable for big project... I'll improve it in feature releases

**Comment by Tom910** [comment]
@smelukov maybe necessary hide node_modules modules. React creates a lot of links

**Comment by lahmatiy** [comment]
I think, hiding of node_modules is not a good solution. Aggregation lib modules in one node looks better approach (so all React modules will represent as single node).

**Comment by Tom910** [comment]
@lahmatiy Then it will be that ~70%+ of the modules are connected with one point. But it will be better than it is now)
Or do as the option to hide and show node_modules

**Comment by lahmatiy** [comment]
Yeah, we should try both options.

**Comment by smelukov** [comment]
Great ideas! Working on it...

**Comment by smelukov** [comment]
The problem with non-usable graph was solved in 1.0 release ;)
